### ISSUE TITLE
`torch.softmax` and cross entropy VJPs violate translation invariance in a finite float64 tail

### ISSUE THREAD

**Issue by beanduan22** [pytorch/pytorch#195531]
### 🐛 Describe the bug

The VJPs returned by `torch.autograd` for `torch.softmax` and
`torch.nn.functional.cross_entropy` do not preserve the translation invariance
of their forward functions in a finite float64 tail.

For any scalar `c`, both functions are invariant under adding `c` to every
logit:

```text
softmax(z + c * 1) = softmax(z)
cross_entropy(z + c * 1, target) = cross_entropy(z, target)
```

Consequently, the gradient with respect to logits must have zero sum.  At
`z = [40.0, 0.0]`, PyTorch returns a nonzero sum of magnitude
`4.248354255291589e-18` for both VJPs.

The expected first component remains finite and representable: for the VJP of
`softmax(z)[0]`, it is approximately `+4.248354255291589e-18`; for the
cross-entropy gradient with target `0`, it is the negative of that value.
PyTorch instead returns `0.0` for the first component while retaining the
opposite tail in the second component.

```python
import math
import torch
import torch.nn.functional as F

torch.set_default_dtype(torch.float64)


def softmax_vjp():
    logits = torch.tensor([40.0, 0.0], requires_grad=True)
    value = torch.softmax(logits, dim=0)[0]
    gradient, = torch.autograd.grad(value, logits)
    print("softmax output:", value.item())
    print("softmax VJP:", gradient.tolist())
    print("softmax VJP sum:", gradient.sum().item())


def cross_entropy_vjp():
    logits = torch.tensor([[40.0, 0.0]], requires_grad=True)
    loss = F.cross_entropy(logits, torch.tensor([0]))
    gradient, = torch.autograd.grad(loss, logits)
    print("cross entropy loss:", loss.item())
    print("cross entropy VJP:", gradient[0].tolist())
    print("cross entropy VJP sum:", gradient.sum().item())


softmax_vjp()
cross_entropy_vjp()
print("expected tail magnitude:", math.exp(-40.0) / (1.0 + math.exp(-40.0)) ** 2)
```

## Actual result

On PyTorch `2.11.0+cu128` with CPU execution:

```text
softmax output: 1.0
softmax VJP: [0.0, -4.248354255291589e-18]
softmax VJP sum: -4.248354255291589e-18

cross entropy loss: 0.0
cross entropy VJP: [0.0, 4.248354255291589e-18]
cross entropy VJP sum: 4.248354255291589e-18

expected tail magnitude: 4.248354255291589e-18
```

The forward functions remain exactly unchanged if `[40.0, 0.0]` is shifted to
`[41.0, 1.0]`; the violation appears in the pullback only.

## Expected result

Both VJPs should sum to zero.  The expected softmax VJP is approximately
`[+4.248354255291589e-18, -4.248354255291589e-18]`; the expected
cross-entropy gradient is its negation.

### Versions

PyTorch version: 2.11.0+cu128
Python version: 3.13.5
Default dtype: float64
Execution device: CPU
